src/query/graphite/native/builtin_functions_test.go

```diff
@@ -261,6 +261,38 @@ func TestScale(t *testing.T) {
 	}
 }
 
+
+func TestUseSeriesAbove(t *testing.T) {
+	ctrl := xgomock.NewController(t)
+	defer ctrl.Finish()
+
+	store := storage.NewMockStorage(ctrl)
+	now := time.Now().Truncate(time.Hour)
+	engine := NewEngine(store)
+	startTime := now.Add(-3 * time.Minute)
+	endTime := now.Add(-time.Minute)
+	ctx := common.NewContext(common.ContextOptions{Start: startTime, End: endTime, Engine: engine})
+	defer ctx.Close()
+
+	stepSize := 60000
+	target := "useSeriesAbove(foo.bar.q.zed, -1, 'q', 'g')"
+
+	store.EXPECT().FetchByQuery(gomock.Any(), "foo.bar.q.zed", gomock.Any()).DoAndReturn(
+		buildTestSeriesFn(stepSize, "foo.bar.q.zed"))
+	store.EXPECT().FetchByQuery(gomock.Any(), "foo.bar.g.zed", gomock.Any()).DoAndReturn(
```

Comment:
Mind adding a test where nothing is replaced, and another where multiple values are replaced?

Reply:
added